### PR TITLE
[New] First Time Seen User Auth via DeviceCode Protocol

### PR DESCRIPTION
**Pull request by Samirbous**
https://github.com/elastic/detection-rules/issues/3665

### CODE REVIEW

**Comment by protectionsmachine** [comment]
## Rule: New - Guidelines

These guidelines serve as a reminder set of considerations when proposing a new rule.

### Documentation and Context

- [ ] Detailed description of the rule.
- [ ] List any new fields required in ECS/data sources.
- [ ] Link related issues or PRs.
- [ ] Include references.

### Rule Metadata Checks

- [ ] `creation_date` matches the date of creation PR initially merged.
- [ ] `min_stack_version` should support the widest stack versions.
- [ ] `name` and `description` should be descriptive and not include typos.
- [ ] `query` should be inclusive, not overly exclusive, considering performance for diverse environments. Non ecs fields should be added to `non-ecs-schema.json` if not available in an integration.
- [ ] `min_stack_comments` and `min_stack_version` should be included if the rule is only compatible starting from a specific stack version.
- [ ] `index` pattern should be neither too specific nor too vague, ensuring it accurately matches the relevant data stream (e.g., use logs-endpoint.process-* for process data).
- [ ] `integration` should align with the `index`. If the integration is newly introduced, ensure the manifest, schemas, and `new_rule.yaml` template are updated.
- [ ] `setup` should include the necessary steps to configure the integration.
- [ ] `note` should include any additional information (e.g. Triage and analysis investigation guides, timeline templates).
- [ ] `tags` should be relevant to the threat and align/added to the `EXPECTED_RULE_TAGS` in the definitions.py file.
- [ ] `threat`, `techniques`, and `subtechniques` should map to ATT&CK always if possible.

#### New BBR Rules
- [ ] `building_block_type` should be included if the rule is a building block and the rule should be located in the `rules_building_block` folder.
- [ ] `bypass_bbr_timing` should be included if adding custom lookback timing to the rule.

### Testing and Validation

- [ ] Provide evidence of testing and detecting the expected threat.
- [ ] Check for existence of coverage to prevent duplication.

**Review by imays11 — APPROVED** [review]

File: rules/integrations/azure/credential_access_first_time_seen_device_code_auth.toml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+[rule]
+author = ["Elastic", "Matteo Potito Giorgio"]
+description = """
+Identifies when a user is observed for the first time in the last 14 days authenticating using the deviceCode protocol. The device code authentication flow can be abused by attackers to phish users and steal access tokens to impersonate the victim. By its very nature, device code should only be used when logging in to devices without keyboards, where it is difficult to enter emails and passwords.
+"""
+from = "now-9m"
+index = ["filebeat-*", "logs-azure.signinlogs-*", "logs-azure.activitylogs-*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "First Time Seen User Auth via DeviceCode Protocol"
```
Comment (terrancedejesus) [inline rules/integrations/azure/credential_access_first_time_seen_device_code_auth.toml:16]:
```suggestion
name = "First Occurrence of Entra ID Auth via DeviceCode Protocol"
```

File: rules/integrations/azure/credential_access_first_time_seen_device_code_auth.toml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "Data Source: Azure",
+    "Data Source: Microsoft Entra ID",
+    "Use Case: Identity and Access Audit",
+    "Tactic: Credential Access",
+]
+timestamp_override = "event.ingested"
+type = "new_terms"
+
+query = '''
+ event.dataset:azure.signinlogs and event.outcome:success and azure.signinlogs.properties.authentication_protocol:deviceCode
```
Comment (terrancedejesus) [inline rules/integrations/azure/credential_access_first_time_seen_device_code_auth.toml:38]:
```suggestion
 event.dataset:(azure.activitylogs or azure.signinlogs)
     (azure.signinlogs.properties.authentication_protocol:deviceCode or azure.activitylogs.properties.authentication_protocol:deviceCode)
     and event.outcome:success
```

**Review by terrancedejesus — APPROVED** [review]
Nice addition!
Works well with `Entra ID Device Code Auth with Broker Client` rule.
Suggested name change only to explicitly label the CSP service associated (Entra ID) as we do with the CSP rules. Not a big deal though.
Added 2 additions to the query to account for activitylogs as well.

**Pull request merged**